psi4/src/psi4/lib3index/df_helper.h

```diff
@@ -109,6 +117,14 @@ class DF_Helper {
     void fill_tensor(std::string name, SharedMatrix M, std::pair<size_t, size_t> a1,
       std::pair<size_t, size_t> a2, std::pair<size_t, size_t> a3);
 
+    // only use this one if you now what you're doing!
```

Comment:
This should be a common operation, can you add a little info on what this does here and with the SharedMatrix. Worth commenting on how you fill the the matrix when matrix size is not equal to the tensor slice size.